File: 1027-longest-arithmetic-subsequence/1027-longest-arithmetic-subsequence.py
class Solution:
    def longestArithSeqLength(self, nums: List[int]) -> int:
        n = len(nums)
        dp, res = [{} for _ in range(n)], 0
        for i in range(1, n):
            for j in range(0, i):
                diff = nums[i] - nums[j]
                dp[i][diff] = max(dp[i].get(diff, 0), 1 + dp[j].get(diff, 0))
                res = max(res, 1 + dp[i][diff])
        return res

File: 1027-longest-arithmetic-subsequence/1027-longest-arithmetic-subsequence.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var longestArithSeqLength = function(nums) {
    const n = nums.length
    const dp = Array(n).fill(0).map(() => ({}))
    let max = 0
    for (let i=1; i < n; i++) {
        for (let j=0; j < i; j++) {
            const diff = nums[i] - nums[j]
            dp[i][diff] = Math.max((dp[i][diff] || 0), 1 + (dp[j][diff] || 0))
            max = Math.max(max, 1 + dp[i][diff])
        }
    }
    return max
};